### DISCUSSION TITLE
timeout default response

### DISCUSSION THREAD

**Discussion by andreatrygit** [sindresorhus/ky#352] [Q&A] [answered]
Hi there!
Thanks for ky which is superb!
I'm looking for the cleanest syntax of defaulting a response when my POST times out.
Basically I want retry:0 and timeout:4000.
I suppose that TimeoutError is thrown if server times out, which should be handled with try catch block.
Is there a cleaner way?
Thought with retry:1 and a beforeRetry hook, but returning ky.stop would erase the default...
Any help??
Thanks very much!
Cheers from Italy! 👍👍

**Comment by sholladay** [comment]
Would it help if `beforeRetry` hooks could return a `Response`?

**Reply by andreatrygit** [reply]
yes indeed. I'd craft a default Response and then I'd return it blocking the hook with ky.stop.
Or should I simply return a response?

**Reply by sholladay** [reply]
I've added this as a feature request to #297. You would just return the response and Ky would automatically short circuit the rest of the hooks, so `ky.stop` would not be needed.

(Although, we are discussing whether short-circuiting should be opt-in, see https://github.com/sindresorhus/ky/issues/387.)

**Comment by sholladay** [comment, marked as answer]
See https://github.com/sindresorhus/ky/issues/297 for future development updates.
